src/checkmatelib/client.py

```diff
@@ -14,13 +16,19 @@ class CheckmateClient:
 
     MAX_URL_LENGTH = 2000
 
-    def __init__(self, host):
+    def __init__(self, host, api_key=None):
         """Initialise a client for contacting the Checkmate service.
 
         :param host: The host including scheme, for the Checkmate service
+        :param api_key: Secret to connect to the server.
+            Defaults to CHECKMATE_API_KEY env var if present.
         """
         self._host = host.rstrip("/")
 
+        self.api_key = api_key
```

Comment:
I've seen something like this in for example sentry's library and I think it's quite flexible, for the most general case, just throw a new envvar, if you need something extra pass it the way you want.

Reply:
:+1: